### PR TITLE
feat(range): add ticks props

### PR DESCRIPTION
**Pull request by aesteves60**

### CODE REVIEW

File: packages/ods/src/components/range/src/components/ods-range/ods-range.tsx [hunk 1]
```diff
@@ -24,11 +24,14 @@ export class OdsRange implements OdsFormElement {
...
   private tooltip?: OdsTooltip;
   private tooltipDual?: OdsTooltip;
+  private listId = getRandomHTMLId();
 
   @State() private dualValue?: number;
   @State() private currentValue?: number;
   @State() private isDualRange: boolean = false;
   @State() private isInvalid: boolean | undefined;
+  @State() private parsedTicks: number[] = [];
+  @State() private activatedTicks: number[] = [];
```
Comment (dpellier) [inline packages/ods/src/components/range/src/components/ods-range/ods-range.tsx:34]: heep alphabetical order

File: packages/ods/src/components/range/src/components/ods-range/ods-range.scss [hunk 2]
```diff
@@ -4,6 +4,7 @@
 :host {
   display: inline-flex;
   position: relative;
+  padding-top: 5px;
```
Comment (dpellier) [inline packages/ods/src/components/range/src/components/ods-range/ods-range.scss:7]: should be computed from tick height and range thickness

File: packages/ods/src/components/range/src/components/ods-range/ods-range.scss [hunk 3]
```diff
@@ -44,4 +45,22 @@
...
+  &__ticks {
+    position: absolute;
+    inset: 0 calc(range.$ods-input-range-size-thumb / 2);
+    z-index: range.$range-z-index-dual - 2;
+
+    &__tick {
+      position: absolute;
+      border-radius: 6px;
+      background-color: range.$range-background-color;
+      width: 3px;
```
Comment (dpellier) [inline packages/ods/src/components/range/src/components/ods-range/ods-range.scss:58]: 2px

File: packages/ods/src/components/range/src/components/ods-range/ods-range.scss [hunk 4]
```diff
@@ -44,4 +45,22 @@
...
+      height: 18px;
```
Comment (dpellier) [inline packages/ods/src/components/range/src/components/ods-range/ods-range.scss:59]: should be the same as the thumb, (declare a var that you'll use on the thumb style also)

File: packages/ods/src/style/_range.scss [hunk 5]
```diff
@@ -159,7 +160,7 @@ $range-z-index-dual: 1;
   .ods-range__range {
     position: absolute;
     z-index: $range-z-index-dual + 1;
-    margin-top: 6px;
+    margin-top: 4px;
```
Comment (dpellier) [inline packages/ods/src/style/_range.scss:163]: relative to what?

**Review by dpellier — CHANGES_REQUESTED** [review]

File: packages/ods/src/style/_range.scss [hunk 6]
```diff
@@ -11,12 +11,15 @@ $range-z-index-dual: 1;
...
   appearance: none;
+  margin: 0;
+  margin-top: calc($ods-input-range-size-thumb / 4);
   outline: none;
   border-radius: var(--ods-border-radius-sm);
   cursor: pointer;
   height: $ods-input-range-height;
   pointer-events: none;
 
+
```
Comment (dpellier) [inline packages/ods/src/style/_range.scss:22]: remove empty line

File: packages/ods/src/style/_range.scss (hunk 5 above)
Comment (dpellier) [reply packages/ods/src/style/_range.scss:165]: not event needed? should be overriden by the one declared above anyway

File: packages/ods/src/components/range/src/components/ods-range/ods-range.tsx [hunk 7]
```diff
@@ -288,6 +312,67 @@ export class OdsRange implements OdsFormElement {
...
+
+      <div class="ods-range__ticks">
+        {
+          this.parsedTicks.map((tick) => (
+            <div
+              class={{
+                'ods-range__ticks__tick': true,
+                'ods-range__ticks__tick--activated': this.activatedTicks.indexOf(tick) > -1,
+              }}
+              style={{ left: `calc(${tick * ratio}% - 2px)` }}>
```
Comment (dpellier) [inline packages/ods/src/components/range/src/components/ods-range/ods-range.tsx:336]: this `2` is in fact the half of the tick thickness

**Review by dpellier — CHANGES_REQUESTED** [review]

**Review by dpellier — CHANGES_REQUESTED** [review]
- need to add more space between text and thumb (especially when focused)
Capture d’écran 2024-12-09 à 14 26 09
Capture d’écran 2024-12-09 à 14 26 11


- KO on chromium when activating the ticks
Capture d’écran 2024-12-09 à 14 26 34
Capture d’écran 2024-12-09 à 14 26 43

**Review by dpellier — CHANGES_REQUESTED** [review]
The height of the ods-range element is now incorrect, with the extra space added to the texts.
Capture d’écran 2024-12-10 à 11 02 28

The correct way would be to enforce an height on those texts and use it to compute the ods-range value (with the range height variable)

**Review by dpellier — APPROVED** [review]

**Review by astagnol — CHANGES_REQUESTED** [review]
Active ticks when Range is disabled are incorrect:
Capture d’écran 2024-12-11 à 09 16 58

**Comment by feoche** [comment]
- [] On "Ticks" example (only single ones) in technical documentation, please remove `0` and `100` ticks

**Pull request merged**